packages/gatsby-link/src/index.js

```diff
@@ -73,8 +55,27 @@ class GatsbyLink extends React.Component {
             !e.ctrlKey &&
             !e.shiftKey
           ) {
```

Comment:
How's this looking to you now? :)

If I misunderstood your instructions, let me know and I can adjust further.

Note that I kept `onClick && onClick(e)` outside the if statement so that it still can be overridden if need be.